### DISCUSSION TITLE
[Feature Request] Is there a mojo "strict mode"?

### DISCUSSION THREAD

**Discussion by masters3d** [modular/modular#2536] [Ideas]
I am thinking something similar to JS "[strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode)". Let's imagine I only want my mojo module to be consumed by other mojo modules (of course my consumers can choose to interact with python).  Is there a way to enforce this "strict mode" per module or per file? 

The immediate why for this question is for a human learning mojo (Maybe there are other use cases like performance)

**Comment by ivellapillil** [comment]
There is no strict mode for Mojo. AFAIK, it is not planned either.

**Comment by hogepodge** [comment]
Thanks for the feedback @masters3d. It's not on our roadmap for Mojo, but we appreciate the request. I've converted this conversation to an idea💡 as something for the product team to consider.
